iCubGenova09/hardware/mechanicals/right_arm-eb3-j0_1-mec.xml

```diff
@@ -12,16 +12,16 @@
         <param name="fullscalePWM">             32000               32000               </param>
         <param name="ampsToSensor">             1000.0              1000.0              </param>
         <param name="Gearbox_M2J">              -100.00             -100.00             </param>
-        <param name="Gearbox_E2J">              1                   1                   </param>
+        <param name="Gearbox_E2J">              64                  64                  </param>
         <param name="useMotorSpeedFbk">         1                   1                   </param>
         <param name="MotorType">                "MOOG-BL-C2900576"  "MOOG-BL-C2900576"  </param>
         <param name="Verbose"> 0 </param>
     </group>
     
     <group name="LIMITS">
         <!--                                    0                   1                   -->
-        <param name="hardwareJntPosMax">        25                165                   </param>
-        <param name="hardwareJntPosMin">       -180                 0                   </param>
+        <param name="hardwareJntPosMax">        36.2                165                   </param>
```

Comment:
As above, this limit is a bit weird

Reply:
36.2 is the measured angle of the hard stop; this is necessary to let the incremental procedure work

cc @ale-git

Reply:
As @davidetome said, we measured the back hard stop position after absolute calibration, in order to make the encoder reading the same when switched to hard stop. We did it more than once, so it is not likely (though still possible) that we ran into a miscalibration. Anyway, a difference of 16 degrees must be evident after hard stop calibration... how does the shoulder seem when you put it in 0 position? We checked the 0 and 90 degrees position and it seemed fine. Maybe the limit was reduced for control reasons? In this case only the sotware limit should have been changed, but maybe this wasn't clear at that time. Another possibility is that something wasn't properly mounted, as happened once in the shoulder yaw. I suggest 1) to visually check the 0 and 90 degrees positions and 2) to check the CAD limits.

Reply:
@S-Dafarra , I thought we should check again the limits (both) to see if one increases the other one decreases.. what about @ale-git?

Reply:
Yes you can check if the other limit is ok. Good idea.

Reply:
I checked before on the robot and the hard limit is reached at -180deg and +25deg :thinking:

Reply:
Couldn't it be that @ale-git and @davidetome focused on the lower bound and for some reason it was set to 36.2 deg instead of 25 deg (for the incremental calibration we check against only one hard-stop) and the other upper bound remained unchanged at -180 deg instead of being "increased" to -180+(36.2-25) = -168.8 deg to keep the entire range unaltered?

Ideally and barring all the unfortunate cases of the shoulder not being properly mounted, we should make the lower bound equal to the absolute value of 25 deg and then stick to the admissible range [25, -180] deg.

Reply:
Unfortunately the documentation doesn't help
![image](https://user-images.githubusercontent.com/4729270/152977798-26961169-b245-472e-9452-69a15af94226.png)

Anyway, if we want the back limit at 25 deg all we need to do is to put 25 instead of 36.2 in the calibration file. I strongly suggest to make the quick test of moving the joint by hand to hard stops and see 1) if the upper limit is still -180 and 2) if the arm really reaches the vertical position, since I doubt about this last point.

Reply:
The shoulder pitch has been recently changed by Elio Massa: see https://github.com/icub-tech-iit/tickets/issues/1410#issuecomment-953972983.

Maybe, @ale-git @davidetome you were not aware of this.
And the documentation is sadly not up-to-date, yep.

Reply:
> Anyway, if we want the back limit at 25 deg all we need to do is to put 25 instead of 36.2 in the calibration file. I strongly suggest to make the quick test of moving the joint by hand to hard stops and see 1) if the upper limit is still -180 and 2) if the arm really reaches the vertical position, since I doubt about this last point.

I did exactly this :wink: 

![image](https://user-images.githubusercontent.com/18591940/152986261-1dd17b83-5942-4575-a32c-0f2f85d228e2.png)


I guess also the corresponding parameter in the calibration file (https://github.com/ami-iit/robots-configuration/blob/6b72214dac196be6ea417fa3edb9669839c26e50/iCubGenova09/calibrators/left_arm-calib.xml#L26) should change right?

Reply:
but this is the roll joint.. isn't it?

Reply:
Yes, the 36.2 value must be changed to your actual desired limit.

Reply:
@pattacini I didn't know about the limit upgrade! It seems 170 degrees more than 180 with respect to the vertical axis btw, but the 0 is arbitrary of course :)